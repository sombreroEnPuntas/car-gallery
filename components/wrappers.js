import styled from 'styled-components'

export const fonts = {
  fontFamily: "'Press Start 2P', monospace",
  lineHeight: '1.5em',
}

export const colors = {
  black: {
    background: 'white',
    outline: '"#e7e7e7"',
    shadow: '#212529',
  },
  error: {
    background: 'rgba(231, 110, 85, 0.5)',
    outline: '#ce372b',
    shadow: '#e76e55',
  },
  success: {
    background: 'rgba(146, 204, 65, 0.5)',
    outline: '#76c442',
    shadow: '#92cc41',
  },
}

export const breakpoints = {
  mobile: '768px',
}

export const ErrorBox = styled.pre`
  background-color: ${colors.error.background};
  box-shadow: 4px 0 ${colors.error.shadow}, 0 -4px ${colors.error.shadow},
    -4px 0 ${colors.error.shadow}, 0 4px ${colors.error.shadow};
  display: block;
  margin: 0.5em 0;
  overflow-x: auto;
  padding: 0.5em;
  white-space: pre-wrap;
`

export const ErrorCode = styled.code`
  color: ${colors.error.outline};
  font-family: ${fonts.fontFamily};
  font-size: 0.65em;
  font-weight: bold;
`

export const PageWrapper = styled.div`
  margin: auto;
  width: 96vw;

  @media (min-width: ${breakpoints['mobile']}) {
    width: 80vw;
  }
`

export const FormLineWrap = styled.div`
  display: flex;
  width: 100%;
`

export const Input = styled.input`
  border: none;
  box-shadow: 0 4px ${props => colors[props.status || 'black'].shadow},
    0 -4px ${props => colors[props.status || 'black'].shadow},
    4px 0 ${props => colors[props.status || 'black'].shadow},
    -4px 0 ${props => colors[props.status || 'black'].shadow};
  font-family: ${fonts.fontFamily};
  line-height: calc(2 * ${fonts.lineHeight});
  margin: 4px;
  outline-color: ${props => colors[props.status || 'black'].outline};
  padding: 0.5rem 1rem;
  text-transform: capitalize;
  width: 100%;

  @media (min-width: ${breakpoints['mobile']}) {
    font-size: ${fonts.lineHeight};
    line-height: ${fonts.lineHeight};
  }
`

export const DropdownWrap = styled.div`
  position: relative;
  width: 100%;
  z-index: 1;
`

export const DropdownList = styled.ul`
  background-color: white;
  border: solid 2px;
  border-top: 0;
  box-shadow: 4px 0 ${colors['black'].shadow}, 0 -4px ${colors['black'].shadow},
    -4px 0 ${colors['black'].shadow}, 0 4px ${colors['black'].shadow};
  list-style: none;
  margin: 0 4px;
  max-height: calc(6 * ${fonts.lineHeight});
  overflow-y: hidden;
  padding: 0;
  position: absolute;
  top: 0;
  width: calc(100% - 12px);
`

export const DropdownItem = styled.li`
  align-items: center;
  display: flex;
  height: ${fonts.lineHeight};
  justify-content: center;
  margin: 0;
  overflow: hidden;
  padding: 0;
  text-align: center;
  text-decoration: none;
  text-overflow: ellipsis;
  text-transform: capitalize;
  white-space: nowrap;
  width: 100%;
`
