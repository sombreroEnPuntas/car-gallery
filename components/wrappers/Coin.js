import styled from 'styled-components'
import { colors } from '.'

const Coin = styled.div`
  width: 21px;
  height: 12px;
  background: ${colors.coin.main};
  box-shadow:
    /* 
    * Inner Background Color
    */ 0 -6px 0 -2px
      ${colors.coin.main},
    0 6px 0 -2px ${colors.coin.main}, 0 -12px 0 -4px ${colors.coin.main},
    0 12px 0 -4px ${colors.coin.main}, 0 -18px 0 -8px ${colors.coin.main},
    0 18px 0 -8px ${colors.coin.main}, 0 -24px 0 -12px ${colors.coin.main},
    0 24px 0 -12px ${colors.coin.main},
    /* 
    * Inner Border
    */ -2px 0 0 0 ${colors.coin.border},
    2px 0 0 0 ${colors.coin.border}, 0 -4px 0 0 ${colors.coin.border},
    0 4px 0 0 ${colors.coin.border}, 0 -10px 0 -2px ${colors.coin.border},
    0 10px 0 -2px ${colors.coin.border}, 0 -14px 0 -4px ${colors.coin.border},
    0 14px 0 -4px ${colors.coin.border}, 0 -20px 0 -8px ${colors.coin.border},
    0 20px 0 -8px ${colors.coin.border}, 0 -26px 0 -12px ${colors.coin.border},
    0 26px 0 -12px ${colors.coin.border},
    /* 
    * Middle Border 
    */ -4px 0 0 0 ${colors.coin.accent},
    4px 0 0 0 ${colors.coin.accent}, 0 -2px 0 2px ${colors.coin.accent},
    0 2px 0 2px ${colors.coin.accent}, 0 -8px 0 0px ${colors.coin.accent},
    0 8px 0 0px ${colors.coin.accent}, 0 -12px 0 -2px ${colors.coin.accent},
    0 12px 0 -2px ${colors.coin.accent}, 0 -16px 0 -4px ${colors.coin.accent},
    0 16px 0 -4px ${colors.coin.accent}, 0 -22px 0 -8px ${colors.coin.accent},
    0 22px 0 -8px ${colors.coin.accent}, 0 -28px 0 -12px ${colors.coin.accent},
    0 28px 0 -12px ${colors.coin.accent},
    /*
    * Outer Border
    */ -6px 0 0 0 ${colors.coin.border},
    6px 0 0 0 ${colors.coin.border}, 0 0 0 4px ${colors.coin.border},
    0 -6px 0 2px ${colors.coin.border}, 0 6px 0 2px ${colors.coin.border},
    0 -10px 0 0 ${colors.coin.border}, 0 10px 0 0 ${colors.coin.border},
    0 -14px 0 -2px ${colors.coin.border}, 0 14px 0 -2px ${colors.coin.border},
    0 -18px 0 -4px ${colors.coin.border}, 0 18px 0 -4px ${colors.coin.border},
    0 -24px 0 -8px ${colors.coin.border}, 0 24px 0 -8px ${colors.coin.border},
    0 -30px 0 -12px ${colors.coin.border}, 0 30px 0 -12px ${colors.coin.border};
`

export default Coin
