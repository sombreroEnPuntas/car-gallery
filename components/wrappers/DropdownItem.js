import styled from 'styled-components'
import { fonts } from '.'

export const DropdownItem = styled.li`
  align-items: center;
  display: flex;
  height: ${fonts.lineHeight};
  justify-content: center;
  margin: 0;
  overflow: hidden;
  padding: 0;
  text-align: center;
  text-decoration: none;
  text-overflow: ellipsis;
  text-transform: capitalize;
  white-space: nowrap;
  width: 100%;
`

export default DropdownItem
