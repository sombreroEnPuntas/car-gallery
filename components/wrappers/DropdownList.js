import styled from 'styled-components'
import { colors, fonts } from '.'

const DropdownList = styled.ul`
  background-color: white;
  border: solid 2px;
  border-top: 0;
  box-shadow: 4px 0 ${colors['black'].shadow}, 0 -4px ${colors['black'].shadow},
    -4px 0 ${colors['black'].shadow}, 0 4px ${colors['black'].shadow};
  list-style: none;
  margin: 0 4px;
  max-height: calc(6 * ${fonts.lineHeight});
  overflow-y: hidden;
  padding: 0;
  position: absolute;
  top: 0;
  width: calc(100% - 12px);
`

export default DropdownList
