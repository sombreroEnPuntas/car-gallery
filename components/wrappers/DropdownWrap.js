import styled from 'styled-components'

const DropdownWrap = styled.div`
  position: relative;
  width: 100%;
  z-index: 1;
`

export default DropdownWrap
