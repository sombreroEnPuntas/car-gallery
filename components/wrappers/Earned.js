import styled from 'styled-components'
import { animations } from '.'

const Earned = styled.div`
  animation: ${animations.rotateIn} 1.4s ease-out forwards,
    ${animations.fadeOut} 2s ease-out forwards;

  position: absolute;
`

export default Earned
