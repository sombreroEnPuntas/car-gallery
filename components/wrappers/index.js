import { keyframes } from 'styled-components'

export const animations = {
  blink: keyframes`
  0% {
    opacity: .2;
  }
  20% {
    opacity: 1;
  }
  100% {
    opacity: .2;
  }
  `,
  rotateIn: keyframes`
  0% {
    opacity: 0;
    transform: rotateY(0) translateY(100%);
  }
  100% {
    opacity: 1;
    transform: rotateY(1800deg) translateY(0);
  }
  `,
  fadeOut: keyframes`
  70% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }`,
}

export const breakpoints = {
  mobile: '768px',
}

export const colors = {
  coin: { main: 'gold', border: 'black', accent: 'goldenrod' },
  black: {
    background: 'white',
    outline: '"#e7e7e7"',
    shadow: '#212529',
  },
  error: {
    background: 'rgba(231, 110, 85, 0.5)',
    outline: '#ce372b',
    shadow: '#e76e55',
  },
  success: {
    background: 'rgba(146, 204, 65, 0.5)',
    outline: '#76c442',
    shadow: '#92cc41',
  },
}

export const fonts = {
  fontFamily: "'Press Start 2P', monospace",
  lineHeight: '1.5em',
}
