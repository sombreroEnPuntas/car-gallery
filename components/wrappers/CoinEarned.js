import styled from 'styled-components'
import { animations } from '.'
import Coin from './Coin'

const CoinEarned = styled(Coin)`
  animation: ${animations.rotateIn} 1.4s ease-out forwards,
    ${animations.fadeOut} 2s ease-out forwards;

  position: absolute;
`

export default CoinEarned
