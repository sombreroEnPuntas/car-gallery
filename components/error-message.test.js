// @flow
import React from 'react'
import { shallow } from 'enzyme'

// Dependencies
import { ErrorCode } from './wrappers'

// Tested Unit
import TestedComponent from './error-message'

const getProps = customProps => ({
  message: 'Hello world!',
  ...customProps,
})

describe('ErrorMessage', () => {
  const wrapper = shallow(<TestedComponent {...getProps()} />)

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('shows "Hello world!"', () => {
    const actual = wrapper.find(ErrorCode).text()
    const expected = 'Hello world!'

    expect(actual).toEqual(expected)
  })
})
