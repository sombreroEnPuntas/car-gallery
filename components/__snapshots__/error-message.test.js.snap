// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches the snapshot 1`] = `
<wrappers__PageWrapper>
  <h3>
    Ooops...
  </h3>
  <p>
    Something went sideways :$
  </p>
  <wrappers__ErrorBox>
    <wrappers__ErrorCode>
      Hello world!
    </wrappers__ErrorCode>
  </wrappers__ErrorBox>
</wrappers__PageWrapper>
`;
